inkcpp/include/globals.h

```diff
@@ -4,15 +4,43 @@
 
 namespace ink::runtime
 {
+	template<typename T>
+	struct ret {
+		operator T() { return value; }
+		operator bool() { return has_value; }
+		T value;
+		bool has_value;
+	};
 	/**
 	* Represents a global store to be shared amongst ink runners.
 	* Stores global variable values, visit counts, turn counts, etc.
 	*/
-	class globals_interface
+	class globals
 	{
 	public:
 		// No public interface yet
 		virtual void dummy() = 0;
```

Comment:
I think you can scrap this function. I believe I only added it because the class needed at least one virtual function to be considered virtual or something.